### PR TITLE
Changes for adding benchmark compare feature

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
This PR:
* Adds new job `compare-benchmarks` to run benchmark comparison between baseline and target run. 
* Updated existing `benchmark-pull-request` job to call `compare-benchmarks` job once it is successfully finished to also update the comparison results on the calling pull_request. 

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/14826.  

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch/benchmark-pull-request.jenkinsfile [hunk 1]
```diff
@@ -278,9 +279,29 @@ pipeline {
...
+                                        // Trigger compare-benchmarks job
+                                            build job: 'compare-benchmarks',
+                                                parameters: [
+                                                    string(name:'BASELINE_TEST_EXECUTION_ID', value: "${benchmarkTestIds.baseline}"),
+                                                    string(name:'BASELINE_TEST_EXECUTION_ID', value: "${benchmarkTestIds.contender}"),
+                                                    string(name:'PULL_REQUEST_NUMBER', value: "${pull_request_number}"),
+                                                    string(name:'REPOSITORY', value: "${repository}")
+                                                ],
+                                                wait: false,
+                                                propagate: false
```
Comment (peterzhuamazon) [inline jenkins/opensearch/benchmark-pull-request.jenkinsfile:303]: Spacing.

**Review by peterzhuamazon — COMMENTED** [review]
1. If someone triggers a benchmark in PR, seems like compare will auto trigger if success?
2. benchmark has 24hr timeout while compare also have 24hr timeout, do we need this long period of time to run them?

**Comment by rishabh6788** [comment]
> 1. If someone triggers a benchmark in PR, seems like compare will auto trigger if success?
> 2. benchmark has 24hr timeout while compare also have 24hr timeout, do we need this long period of time to run them?

1. Yes. 
2. The benchmark job for certain workloads runs for 12-hr, and as the workloads increase in size can go up. But compare job should not take more than 10-15-mins to run, reduced timeout to 1-hr.

File: jenkins/opensearch/benchmark-pull-request.jenkinsfile [hunk 2]
```diff
@@ -278,9 +279,29 @@ pipeline {
...
+                                }
+                                if ("${baseline_cluster_config}" != '') {
+                                        Map <String, String> benchmarkTestIds = getCompareBenchmarkIds(
+                                                                                   baselineClusterConfig: "${baseline_cluster_config}",
+                                                                                   distributionVersion: "${DISTRIBUTION_VERSION}-SNAPSHOT",
+                                                                                   workload: "${TEST_WORKLOAD}",
+                                                                                   pullRequestNumber: "${pull_request_number}"
+                                                                                )
+                                        // Trigger compare-benchmarks job
+                                        build job: 'compare-benchmarks',
```
Comment (peterzhuamazon) [inline jenkins/opensearch/benchmark-pull-request.jenkinsfile:295]: Nit: Possible of changing this to `benchmark-compare` to in sync with other benchmark workflow naming convention?

Thanks.

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**